### DISCUSSION TITLE
Possibility for a more in deept tutorial on how to do everything, specialy how to configure MQTT with the ESP Home  and the Awox lights

### DISCUSSION THREAD

**Discussion by OkOtChA666** [fsaris/EspHome-AwoX-BLE-mesh-hub#66] [General]
I think i've manage to do the instalation on the ESP (I think).  Howver devices are not show on MQTT. Maybe a full tutorial?

**Comment by OkOtChA666** [comment]
OK, i've seen the logs

INFO ESPHome 2023.12.9
INFO Reading configuration /config/esphome/esphome-web-b68a10.yaml...
WARNING GPIO2 is a strapping PIN and should only be used for I/O with care.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
INFO Starting log output from esphome-web-b68a10/debug
INFO Connected to MQTT broker!

What am I doing wrong?

**Reply by fsaris** [reply]
Your board has a different pin layout for the status led. The easiest way to remove this warning is by removing these lines from your yaml.

```
status_led:
  # Match this with the board your using
  pin: GPIO2
```

> INFO Connected to MQTT broker!

Are you sure the wifi and MQTT settings are correct?

What log output do you get when you connect your board using a wired connection?

"Logs" -> "Plug into this computer"

**Comment by OkOtChA666** [comment]
More info
![image](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/assets/92456100/031665aa-693c-4b90-b031-5ff296306bc8)

**Reply by fsaris** [reply]
This is a build warning about unused code, you can ignore this

**Comment by fsaris** [comment]
1. First get familiar with the basics of ESPHome https://esphome.io/guides/getting_started_hassio.html
  to ensure:
   - flashing/installing of a board/esp works
   - the board correctly connects to your wifi

2. Enable logging and validate you can see the logs when the device is running  (details about logging see https://esphome.io/components/logger)
   - Add `logger:` to your example yaml from step 1
   - install the updated yaml config 
   - From Esphome dashboard click on "Logs" ->  "wireless"
   - validate you see the debug logging. It should report info about WiFi connection 
 
3. Switch framework from `arduino` to `esp-idf` 
    - Update the `framework:` `type:` value in the yaml from `arduino` to `esp-idf`
    - install the updated yaml config
    - validate it works by checking the logs again from previous step

4. Setup and test the mqtt connection
   - setup a mqtt broker 
   - remove the `api:` from your test yaml from step 1 by the mqtt config see https://esphome.io/components/mqtt.html

5.  Check if you can also see the logging now you switched to MQTT
    - validate the logs like you did in step 2
    - It should report info about MQTT and WiFi connection 

6. Now the build/install process and connections are working you can move on to adding the AwoX component 
    - bee sure to remove the following line from your yaml (to prevent memory issues)
       - `captive_portal:`
    - add the `external_components:` and all below from the example yaml to your yaml https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/blob/main/awox-ble-mesh-hub.yaml
    - install the updated yaml config
    - validate it works by checking the logs again from step 2
    - the logs should report all found devices etc
    - when devices are found and it could connect to one these will be reported to MQTT (and should show up in Home Assistant)

7. Next update the `awox_mesh:` `device_info:` in your yaml. https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/tree/main#devices

**Comment by OkOtChA666** [comment]
Thanks. Will try this tomorrow

**Comment by OkOtChA666** [comment]
Hey fsaris

So i've done evertything by steps to see what I was doing wrong
One part of code compile... another part of code compile, etc...

I think I made some progress, this is how it's going.

![Win](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/assets/92456100/7584861b-e619-4d1a-9176-db73e1c8cbee)

However when I place the credentials code (the only part of the code that is missing)

awox_mesh:
  mesh_name: !secret mesh_name
  mesh_password: !secret mesh_password

i got the error

![Lose](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/assets/92456100/9f99c6bc-4be8-4cd8-a381-874d216932b3)

MQTT discovers some things (without the AWOX credentials), but I belive is only the ESP board

Advise?

**Reply by fsaris** [reply]
can you post your full `yaml` file here?

**Reply by fsaris** [reply]
and can you "Clean build files" (one of the options in the context menu when you click the 3dots on the card) and try to build again. And post the full build log here.

**Reply by OkOtChA666** [reply]
I did what you said about the Clean build files and it "worked" Ir discoverd some lights
However it seams like its rebooting is this normal...

Here is the log just after the "clean build files"

[Log.txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/14323078/Log.txt)

**Comment by OkOtChA666** [comment]
![Devices](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/assets/92456100/dabfa0ee-b2f7-4bad-8833-43e6bfe6048f)

The messages is as follows:
Last Boot was an unhandled reset, will proceed to safe mode in 8 restarts

And it won't stop

**Reply by fsaris** [reply]
please post you whole `yaml`

looks like a memory issue.

**Comment by OkOtChA666** [comment]
Sure..
[awox-ble-mesh-hub.txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/14323567/awox-ble-mesh-hub.txt)

And thank you for helping

EDIT: Here is the logs from the ESP
[logs_awox-ble-mesh-hub_logs.txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/14323576/logs_awox-ble-mesh-hub_logs.txt)

**Reply by fsaris** [reply]
okay, nothing strange there. What board are you using?

**Reply by fsaris** [reply]
And add some more debugging. Maybe that gives more insight.

This will add some sensors to HA.

```
debug:
  update_interval: 5s

text_sensor:
  - platform: debug
    reset_reason:
      name: "Reset Reason"

sensor:
  - platform: debug
    free:
      name: "Heap Free"
    loop_time:
      name: "Loop Time"
```

**Reply by OkOtChA666** [reply]
NodeMCU ESP32 (Wifi+Bluetooth+SRAM) - JOY-IT

https://mauser.pt/catalog/product_info.php?products_id=096-7620

**Reply by OkOtChA666** [reply]
Reset Reason: Software Reset CPU

[logs_awox-ble-mesh-hub_logs.txt](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub/files/14323668/logs_awox-ble-mesh-hub_logs.txt)

**Reply by fsaris** [reply]
Do you have a different device to test?

you setup looks similar to mine (I have 15 lights). Maybe there is something with the memory of your device. 

Also try to experiment with moving the device. Maybe it react a little bit different when further away from other devices and/or closer to 1 of the lights.

**Comment by OkOtChA666** [comment]
Going to stop for today.
Will try again in a few days, nevertheless thank you for your help.
Just one more question though,  I've notice on your YAML file that the devices ID are something like "0x25".

How do you know the devices ID? I only see the MAC address

Thanks

**Reply by fsaris** [reply]
> How do you know the devices ID? I only see the MAC address

Once the lights are connected it will be visible in the logs (and in HA).

**Comment by OkOtChA666** [comment]
Hello
I did a bit more testing, and it dosent bootloop if i remove the code:

awox_mesh:
  mesh_name: !secret mesh_name
  mesh_password: !secret mesh_password

I have the Eglo Connect Plus which is like a "hub". Do you think its relevant?
